plugin/workspace.vim

```diff
@@ -14,9 +14,33 @@ let g:workspace_autosave_untrailspaces = get(g:, 'workspace_autosave_untrailspac
 let g:workspace_autosave_au_updatetime = get(g:, 'workspace_autosave_au_updatetime', 3)
 let g:workspace_autocreate = get(g:, 'workspace_autocreate', 0)
 let g:workspace_nocompatible = get(g:, 'workspace_nocompatible', 1)
+let g:workspace_session_directory = get(g:, 'workspace_session_directory', '')
+
+function! s:IsSessionDirectoryUsed()
+  return !empty(g:workspace_session_directory)
+endfunction
+
+function! s:GetSessionDirectoryPath()
+	let l:sessionDirectory = g:workspace_session_directory
+  if !isdirectory(g:workspace_session_directory)
+    call mkdir(g:workspace_session_directory)
+  endif
+	let l:cwd = getcwd()
+	let l:fileName = substitute(l:cwd, "/", '-', "g")
```

Comment:
What are your thoughts on changing this to a hash of the string?
ex. `return l:sessionDirectory . sha256(getcwd())`

Pros:
- If the current working directory is ever over 255 characters (which is the limit in most file systems https://en.wikipedia.org/wiki/Filename), then a hash will be consistent length and the plugin won't error when trying to save.
- Avoid collisions between similar paths, such as `/Users/Home/project/configs` and `/Users/Home/project-configs` (both will evaluated to `-Users-Home-project-configs`, if someone decides to create two workspaces, one nested in the other).
- Any weird characters in path will be hashed away!
Cons:
- The filenames will no longer be readable.
- The likelihood of collision is probably really really really low.

What about... maybe 2 files? One for the session directory, one for the actual session data? Ex:
```
8B7DF143D91C716ECFA5FC1730022F6B421B05CEDEE8FD52B1FC65A96030AD52.session.directory
8B7DF143D91C716ECFA5FC1730022F6B421B05CEDEE8FD52B1FC65A96030AD52.session.vim
```

Reply:
I think it's a good idea, might be something for users to opt-in for though.  In the event that their session somehow gets in a weird state it would be nice to be able to blow it away, and having the a readable session name is nice.